### ISSUE TITLE
Consider changing the way how request and response can be extended

### ISSUE THREAD

**Issue by Alannek1992** [hoangvvo/next-connect#172]
The current solution seems to be not sufficient in case that we have more HTTP methods to handle in our API route.

At the moment I can define the shape of the Request and the Response e.g. like this:
`
const handler = nc<NextApiRequest, NextApiResponse>();
`

This basically gives me the basic Next Types for request and response. I could of course extend them:
`
interface PostRequestIf extends NextApiRequest {
  body: {
    email: string;
    name: string;
    message: string;
  };
}
`
Then I can use that like this:
`
const handler = nc<PostRequestIf , NextApiResponse>();
`
This seems to be fairly good enough for the API route, which is handling only the POST request. Imagine that I need to handle also the GET request, in that case, the PostRequestIf does not make sense. It can be handled by type union for instance. But this brings unnecessary complexity to the table including some casting and so on.

The problem is that I cannot specify concretely on the level of GET or POST handler, what is the expected type since the definition is following:
`
    post<ReqExt = {}, ResExt = {}>(
      ...handlers: RequestHandler<Req & ReqExt, Res & ResExt>[]
    ): this;
`
Basically, it says that I can only do an intersection and since the body for the request is defined as any I cannot easily change that (the same applies for the Response):

![image](https://user-images.githubusercontent.com/41963447/148526206-181604c2-87b1-4d1d-a6cb-166df851e43c.png)

IMHO It must be possible to change that definition on the level of the concrete HTTP handling function. Something like this:
`
   post<ReqExt extends Req = {}, ResExt extends Res = {}>(
      ...handlers: RequestHandler<ReqExt, ResExt>[]
    ): this;
`

This gives a typescript developer much more flexibility, could you consider that?

BTW: I love this library anyway <3

Thanks!

**Comment by hoangvvo** [comment]
Sorry for the late response! My final verdict is in https://github.com/hoangvvo/next-connect/pull/171#issuecomment-1024980113.

The only for now is to omitting `body` as the base:

```ts
nc<Omit<NextAPIRequest, "body">, NextAPIResponse>();
```

In which case, we can freely set it to anything later.
